### ISSUE TITLE
Provide a means to opt-in to deferred transactions by default for Microsoft.Data.Sqlite

### ISSUE THREAD

**Issue by p-m-j** [dotnet/efcore#28172]
## What problem are you trying to solve?

Passing a deferred parameter to `BeginTransaction` is not posible when working at a higher level of abstraction i.e. `System.Data.Common.DbConnection` without making use of a decorator.

This makes the provider awkward to work with for any libraries which wish to enable concurrent queries unless they have visibility of a `SqliteConnection`.

## Describe the solution you'd like

Any means by which a flag can be set that opts into deferred transactions by default without causing behavioural breaking changes for users who rely on the existing functionality e.g. a ConnectionString Keyword which when left unset results in the current behaviour.

**Comment by ajcvickers** [comment]
@p-m-j This is not something we plan to add to any higher level of abstraction. Beyond it not being generally supported, deferred transactions are a common source of deadlocks in Sqlite, and hence it isn't something we encourage people to use.

**Comment by p-m-j** [comment]
Deferred transactions are intended to be the default behaviour when using SQLite - https://www.sqlite.org/lang_transaction.html

When in rollback journal mode one should expect to be able have a single write transaction **or** $n$ read transactions executing in parallel.

When in WAL journal mode one should expect to be able have a single write transaction **and** $n$ read transactions executing in parallel.

This was even tried as the default behaviour in 2.x but reverted due to the deadlocks downstream users encountered as you alluded to above e.g. https://github.com/aspnet/Microsoft.Data.Sqlite/issues/474

I think it's fair to say we want to keep existing applications working without forcing a refactor to avoid concurrency issues, but it also seems reasonable that those building new applications/libraries should be able to decide to opt into enabling  concurrency without having to hack around the upstream.

**Comment by ajcvickers** [comment]
Note from triage: on further reflection, we believe that this mode is a pit-of-failure which will very easily result in deadlocks, so we don't want to advertise its use by adding connection string support.
